proxy/mempool/mempool.py

```diff
@@ -523,6 +524,13 @@ def on_gas_price(self, gas_price: MPGasPriceResult) -> None:
     def _create_kick_tx_schedule_task(self) -> None:
         asyncio.get_event_loop().create_task(self._kick_tx_schedule())
 
+    async def _process_tx_eviction_loop(self) -> None:
+        while True:
+            for tx_schedule in self._tx_schedule_dict.values():
+                tx_schedule._drop_old_txs()
+
+            await asyncio.sleep(self._config.mempool_eviction_time_sec)
```

Comment:
If the eviction time is 3 hours, I can send a tx that can live in the mempool <=6 hours.

- loop is started
- loop is finished
- mempool receives a tx
- loop is started
- loop skips tx, because the heartbeat is less than 3 hours
- loop is finished

Reply:
In `_drop_old_txs(self)` we have:

    tx = self._tx_dict.peek_heartbeat_tx()

Here, `peek_heartbeat_tx()` is a sorted queue, so should return older transactions before newer transactions.

So in step 3 of your example above, when the mempool receives a new tx, yes it will be skipped because this individual transaction has a heartbeat is less than 3 hours. In other words, we should not be evicting new transactions, only old ones.